src/lib/BindToStores.react.js

```diff
@@ -1,47 +1,121 @@
+import invariant from 'invariant'
 import React from 'react'
-import StorePrototype from './StorePrototype'
 
-// let's say for now, stores is a dictionary of name and object
-export default function bindResources(Component, resourceName) {
-  function extractStores() { return Object.getOwnPropertyNames(stores).map(store => stores[store]) }
-  function capitalize(word) { return word.charAt(0).toUpperCase() + word.slice(1) }
-  function singularize(word) { return (word.slice(-1) === 's' ? word.slice(0, -1) : word) }
+import Container from '../lib/Container'
+import storePrototype from './StorePrototype'
 
-  var bootAction = `fetch${capitalize(resourceName)}`
+export default function bindResources(Component, resources, onBoundUpdate = null, resourceId = null) {
+  const stores = {}
 
-  // figure out how to pass in actions
-  // import actionClass from `PATH TO ACTIONS/${singularize(capitalize(resourceName))}Actions`
+  function capitalize(word, pascal = true) {
+    return (pascal ? word.charAt(0).toUpperCase() : word.charAt(0)) + word.slice(1).replace(/(\_\w)/g, match => match[1].toUpperCase())
```

Comment:
Note the new `pascal` param that will by default capitalize the first letter of the word too.